### DISCUSSION TITLE
Quality Engineering Blog

### DISCUSSION THREAD

**Discussion by dorin100** [input-output-hk/quality-engineering#6] [Q&A]
What would be the best tool to use for a Chapter blog?

Some questions:
- who will be the audience? probably QE Chapter + whole IOG
- what functionalities the blog should offer?
  - search option
  - labels for articles
  - anything else? 
- visibility - probably private in the beginning (only for IOG)
  - maybe we might use the Wiki section of this repo (that is already private)?

**Comment by antonbaliasnikov** [comment]
One of the options can be to create a QA Chapter portal with some guides, tutorials, and a blog.

We can use https://docusaurus.io/ for all of this or something similar.

**Reply by emiride** [reply]
I wanted to suggest free version of stackoverflow for teams, but this docusaurus looks very nice. I think https://playwright.dev is created using this tool? The only problem with this is that we have to take care of hosting it and maintaining it. Maybe it is not a problem?

**Reply by sp-io** [reply]
Docusaurus setup is pretty straightforward, yet powerful. @martinseanhunt could do a hosting setup for us, and we won't need to maintain it then. He did the same for Midnight.

**Comment by alexromanov** [comment]
1. We can use Confluence. 
2. We can use a static github.io site from GitHub to build the documentation from the current repo

**Reply by sp-io** [reply]
What I miss on confluence is 'code review' process which Docusaurus setup gives.

**Reply by dorin100** [reply]
A static github.io site would allow a code review option. And we already have this private repo, so it should be private and available only within the IOG org.

**Reply by mkoura** [reply]
I believe github pages are publicly available even for private repositories.

**Reply by mkoura** [reply]
https://docs.github.com/en/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https

> Warning: GitHub Pages sites are publicly available on the internet, even if the repository for the site is private.

**Comment by rsporny** [comment]
Before we decide which tool to use, I'd rather like to have a general overview of what type of content are we going to have/produce in QE Chapter. I assume blog will be just a part of that, so we should try to keep it simple, to not end up with many different places/tools. Moreover, we should define the purpose of blog, what kind of content it's gonna have, and what audience are we targeting.

**Reply by dorin100** [reply]
Excellent point! Let's think about it.

Some usages for a Blog:
- Education - sharing of best practices, test approaches, frameworks, results reporting tools 
- Show and Tell 
- Results of some experiments
- Increase the flow of ideas inside the Chapter 
- Increase the trust in the QE Chapter (team members both inside the company at also in our teams/squads/tribes)
- Increase the knowledge sharing and flow of ideas inside the chapter 

what else !?

**Reply by dorin100** [reply]
Other types of content inside the QE Chapter I can think of:
- results reporting at Squad/Trive/Chapter level - per release, nightly, etc
- test strategies - per tribe/feature
- metrics - test, quality, etc (to be defined)
- processes

**Reply by dorin100** [reply]
here is a blog example (for usage, topics, etc) - https://testing.googleblog.com/

**Reply by sp-io** [reply]
And Docusaurus gives blogging option too...

**Comment by alexromanov** [comment]
For a free setup, we can use [Hugo](https://gohugo.io/) as a template engine and some plugins, like [papermod](https://github.com/adityatelange/hugo-PaperMod). The only thing that we need here is hosting on some domains like testing.iog.io. 
As for visibility - we need to think about it in advance. If we plan to make it public, we need to publish posts according to our public doc guidelines (without any internals, etc).

**Reply by mkoura** [reply]
Hugo looks good! And it is already packaged in many Linux distributions and repositories (including nix), which is nice bonus.
In the end, anything that can build static pages from Markdown will suffice. We can host it on github pages, maybe with private domain (administrator of iog.io would just need to setup DNS records for us).

**Comment by dorin100** [comment]
> The only thing that we need here is hosting on some domains like testing.iog.io.

Can we already use something that would not require hosting but at the same time still be in our control?

**Reply by sp-io** [reply]
Ask @martinseanhunt for best out-of-the-box solution proposal

**Reply by dorin100** [reply]
@martinseanhunt is proposing https://vercel.com/templates/next.js/tailwind-css-starter-blog

**Reply by angerman** [reply]
engineering.IOG.io as well as the cardano updates both use GitHub and Docusaurus actions.

**Comment by dnadales** [comment]
What about using the same framework we use in https://github.com/input-output-hk/cardano-updates, Hydra and Consensus.

**Reply by dorin100** [reply]
can we make that private (only IOG)?

**Reply by dnadales** [reply]
I don't know. Can you remind me why we want this to be private?

**Reply by dorin100** [reply]
We might touch on topics that are related to projects or repositories that are not yet open source, or we might discuss experimental or sensitive info that could disclose IOG's confidential info. 

As for the blog's exposure, the question of whether it should be public or internal is still up for debate. Personally, I don't perceive a substantial benefit in making it accessible to the public as opposed to keeping it in-house.

**Reply by dnadales** [reply]
For sensitive information maybe a Wiki in a private repo might be enough. I'd choose that because people are already on GitHub and we don't need to spend money or people's time on additional tools.

**Comment by dorin100** [comment]
IT is suggesting Confluence Blog functionality.

**Reply by angerman** [reply]
I really would like not to use any Atlassian products if I can help it.

**Reply by coot** [reply]
Why not use https://github.com/input-output-hk/engineering?

**Reply by coot** [reply]
I know this isn't an answer for the original question, but we have both @dorin100 & @angerman in this thread :).

@dorin100 the type of content you mentioned in another thread, fits quite well into the engineering blog that we already have. `docusourus` allows one to have different topics, which makes it easy to separate the content into different categories.  We use it in `cardano-updates`.

**Reply by antonbaliasnikov** [reply]
It's the Docusaurus variant I initially proposed [here](https://github.com/input-output-hk/quality_engineering/discussions/6#discussioncomment-4861224). We can have "Quality" tab.

**Reply by sp-io** [reply]
I don't think adding any new fancy tech stack gives that much benefit. Docusaurus alike stack is used already in cardano docs, so why not use it as it is. Simple, yet powerful. And we can have setup pretty out of the box, with pre-prod environment. The content is more important here, not the tech stack.

**Reply by dorin100** [reply]
> I know this isn't an answer for the original question, but we have both @dorin100 & @angerman in this thread :).
> 
> @dorin100 the type of content you mentioned in another thread, fits quite well into the engineering blog that we already have. `docusourus` allows one to have different topics, which makes it easy to separate the content into different categories. We use it in `cardano-updates`.

I feel the public IOG engineering blog is already a very good tool to share info with the outside.

The Quality Engineering Blog should be used for **internal** alignment and education on different topics thus it should be private.
